cardano-api/cli/generateconfigs/skyline_params.go

```diff
@@ -0,0 +1,488 @@
+	cardanoBlockfrostAPIKey        string
+	cardanoSocketPath              string
+	cardanoTTLSlotInc              uint64
+
+	logsPath         string
+	utxoCacheTimeout time.Duration
+	utxoCacheKeys    []string
+
+	oracleAPIURL string
+	oracleAPIKey string
+
+	apiPort uint32
+	apiKeys []string
+
+	outputDir      string
+	outputFileName string
+
+	primeCardanoWrappedTokenName string
+	cardanoPrimeWrappedTokenName string
+}
+
+func (p *skylineGenerateConfigsParams) validateFlags() error {
+	err := validateAddress(
+		true, p.primeBridgingAddress, primeBridgingAddressFlag,
+		wallet.CardanoNetworkType(p.primeNetworkID))
+	if err != nil {
+		return err
+	}
+
+	err = validateAddress(
+		true, p.primeBridgingFeeAddress, primeBridgingFeeAddressFlag,
+		wallet.CardanoNetworkType(p.primeNetworkID))
+	if err != nil {
+		return err
+	}
+
+	err = validateAddress(
+		false, p.primeBridgingFallbackAddress, primeBridgingFallbackAddressFlag,
+		wallet.CardanoNetworkType(p.primeNetworkID))
+	if err != nil {
+		return err
+	}
+
+	if p.primeBlockfrostURL == "" && p.primeSocketPath == "" && p.primeOgmiosURL == "" {
+		return fmt.Errorf("specify at least one of: %s, %s, %s",
+			primeBlockfrostURLFlag, primeSocketPathFlag, primeOgmiosURLFlag)
+	}
+
+	if p.primeBlockfrostURL != "" && !common.IsValidHTTPURL(p.primeBlockfrostURL) {
+		return fmt.Errorf("invalid prime blockfrost url: %s", p.primeBlockfrostURL)
+	}
+
+	if p.primeOgmiosURL != "" && !common.IsValidHTTPURL(p.primeOgmiosURL) {
+		return fmt.Errorf("invalid prime ogmios url: %s", p.primeOgmiosURL)
+	}
+
+	err = validateAddress(
+		true, p.cardanoBridgingAddress, cardanoBridgingAddressFlag,
+		wallet.CardanoNetworkType(p.cardanoNetworkID))
+	if err != nil {
+		return err
+	}
+
+	err = validateAddress(
+		true, p.cardanoBridgingFeeAddress, cardanoBridgingFeeAddressFlag,
+		wallet.CardanoNetworkType(p.cardanoNetworkID))
+	if err != nil {
+		return err
+	}
+
+	err = validateAddress(
+		false, p.cardanoBridgingFallbackAddress, cardanoBridgingFallbackAddressFlag,
+		wallet.CardanoNetworkType(p.cardanoNetworkID))
+	if err != nil {
+		return err
+	}
+
+	if p.cardanoBlockfrostURL == "" && p.cardanoSocketPath == "" && p.cardanoOgmiosURL == "" {
+		return fmt.Errorf("specify at least one of: %s, %s, %s",
+			cardanoBlockfrostURLFlag, cardanoSocketPathFlag, cardanoOgmiosURLFlag)
+	}
+
+	if p.cardanoBlockfrostURL != "" && !common.IsValidHTTPURL(p.cardanoBlockfrostURL) {
+		return fmt.Errorf("invalid cardano blockfrost url: %s", p.cardanoBlockfrostURL)
+	}
+
+	if p.cardanoOgmiosURL != "" && !common.IsValidHTTPURL(p.cardanoOgmiosURL) {
+		return fmt.Errorf("invalid cardano ogmios url: %s", p.cardanoOgmiosURL)
+	}
+
+	if !common.IsValidHTTPURL(p.oracleAPIURL) {
+		return fmt.Errorf("invalid oracle API url: %s", p.oracleAPIURL)
+	}
+
+	if p.oracleAPIKey == "" {
+		return fmt.Errorf("missing %s", oracleAPIKeyFlag)
+	}
+
+	if len(p.apiKeys) == 0 {
+		return fmt.Errorf("specify at least one %s", apiKeysFlag)
+	}
+
```

Comment:
missing validations for primeCardanoWrappedTokenName and cardanoPrimeWrappedTokenName - they can be found on apex-bridge